### ISSUE TITLE
Useless check for chars

### ISSUE THREAD

**Issue by xamgore** [rust-lang/rust-clippy#14052]
### Summary

`to_ascii_lowercase` is totally fine.

### Lint Name

clippy::manual_ignore_case_cmp

### Reproducer

I tried this code:

```rust
let x: char = 'a';
if x.to_ascii_lowercase() == 's' { ... }
```

I saw this happen:

```
manual case-insensitive ASCII comparison Help: for further information visit https://rust-lang.github.io/rust-clippy/master/index. html#manual_ignore_case_cmp Note: `#[warn(clippy::manual_ignore_case_cmp)]` on by default Help: consider using `.eq_ignore_ascii_case()` instead
```

```rust
let x: char = 'a';
if x.eq_ignore_ascii_case(&'s') { ... }

pub const fn eq_ignore_ascii_case(&self, other: &char) -> bool {
        self.to_ascii_lowercase() == other.to_ascii_lowercase()
    }
```

### Version

```text
rustc 1.84.0 (9fc6b4312 2025-01-07)
binary: rustc
commit-hash: 9fc6b43126469e3858e2fe86cafb4f0fd5068869
commit-date: 2025-01-07
host: aarch64-apple-darwin
release: 1.84.0
LLVM version: 19.1.5
```

**Comment by xamgore** [comment]
How does comparing strings relate to comparing chars?

**Comment by yotamofek** [comment]
Just ran into this myself.

I'll argue that the for `char`s (and `u8`s, BTW), this lint being categorized in the "perf" group is a FP.
It might be correct as a style lint or even pedantic lint, but for these types, using `eq_ignore_ascii_case` has no perf benefits. `eq_ignore_ascii_case` is actually implemented as `a.to_ascii_lowercase() == b.to_ascii_lowercase()`, so might even have a negative perf effect in @xamgore 's example, where he knows the RHS is lowercase (though I assume this check would be optimized out by the compiler).

Case in point: this lint actually fires on the implementations of `{char|u8}::eq_ignore_ascii_case` when run on the rustc repo.

**Comment by ada4a** [comment]
@yotamofek please do!

**Comment by yotamofek** [comment]
> [@yotamofek](https://github.com/yotamofek) please do!

I'll try to get to it sometime soon, can't promise though :)
